pkg/cmd/build/build.go

```diff
@@ -212,67 +212,172 @@ func translateDockerErr(err error) error {
+	} else if manifestBuildOptions.OutputMode == "" {
+		manifestBuildOptions.OutputMode = setOutputMode(oktetoLog.GetOutputFormat())
 	}
 
-	file := b.Dockerfile
-	if !filepath.IsAbs(b.Dockerfile) && !model.FileExistsAndNotDir(file) {
-		file = filepath.Join(b.Context, b.Dockerfile)
+	if cmdBuildOptions.Tag != "" {
+		manifestBuildOptions.Tag = cmdBuildOptions.Tag
 	}
-	opts := &BuildOptions{
-		CacheFrom:   b.CacheFrom,
-		Target:      b.Target,
-		Path:        b.Context,
-		Tag:         b.Image,
-		File:        file,
-		BuildArgs:   args,
-		ExportCache: b.ExportCache,
+
+	// override Tag if global registry
+	if okteto.Context().IsOkteto && cmdBuildOptions.BuildToGlobal {
+		manifestBuildOptions.Tag = registry.ReplaceTargetRepository(manifestBuildOptions.Tag, okteto.GlobalRegistry, manifestBuildOptions.Namespace)
 	}
 
-	outputMode := oktetoLog.GetOutputFormat()
-	if o != nil && o.OutputMode != "" {
-		outputMode = o.OutputMode
+	return manifestBuildOptions
+}
+
+// OptsFromManifest returns the parsed options for the build from the manifest
+func OptsFromManifest(service string, manifestBuildInfo *model.BuildInfo, cmdBuildOptions *BuildOptions) *BuildOptions {
+	manifestBuildOptions := getBuildOptionsFromManifest(service, manifestBuildInfo)
+	return overrideManifestBuildOptions(manifestBuildOptions, cmdBuildOptions)
+}
+
+// optimizedGlobalBuild returns if build should be optimized because the image is already at the global registry
+func (opts *BuildOptions) optimizedGlobalBuild() (string, bool, error) {
+	oktetoLog.Debugf("Applying global build optimization for image %s", opts.Tag)
+
+	globalReference := opts.Tag
+	if registry.IsDevRegistry(opts.Tag) || registry.IsExtendedOktetoRegistry(opts.Tag) {
+		globalReference = registry.ReplaceTargetRepository(opts.Tag, okteto.GlobalRegistry, okteto.DefaultGlobalNamespace)
+
 	}
-	opts.OutputMode = setOutputMode(outputMode)
+	tagWithDigest, err := registry.GetImageTagWithDigest(globalReference)
+	if err != nil {
+		return opts.Tag, false, err
+	}
+	oktetoLog.Debugf("skipping build: image %s is already built", opts.Tag)
+	opts.Tag = globalReference
+	return tagWithDigest, true, nil
+}
 
-	return opts
+// checkImageAtRegistry returns if build should be optimized because the image is already at the dev registry
+func (opts *BuildOptions) checkImageAtRegistry() (string, bool, error) {
+	oktetoLog.Debugf("Checking registry for image %s", opts.Tag)
+	tagWithDigest, err := registry.GetImageTagWithDigest(opts.Tag)
+	if err != nil {
+		return opts.Tag, false, err
+	}
+	oktetoLog.Debugf("skipping build: image %s is already built", opts.Tag)
+	return tagWithDigest, true, nil
 }
 
-// ShouldOptimizeBuild returns if optimization should be applied
-func ShouldOptimizeBuild(image string) bool {
+// SkipBuild returns if build has to be skipped and the tag with digest if found at registry
+func (opts *BuildOptions) SkipBuild(service string) (string, bool, error) {
```

Comment:
Sounds a bit odd a function called `SkipBuild` which return something besides an error because it actually doesn't skip the build, it is just checking things. `ShouldSkipBuild` sounds better but doesn't match when returning the tag.

Do we really need the tag + the boolean? Wouldn't be enough returning the tag and the error?

Reply:
Maybe the name sould be `FindBuiltImageTag`
Also I would remove the `Skipping '%s' build.` since it should be on the callers of this function